gpii/node_modules/flowManager/test/data/browserChannel_deviceReporter.json

```diff
@@ -1,9 +1,9 @@
 [
     {
-        "id": "org.chrome.cloud4chrome"
+        "id": "org.nvda-project"
```

Comment:
IMO there should never have been any "real" solutions here at all.  As many tests as possible should be using test SR data that does not even remotely resemble the live data.   Ideally the test data would be namespaced for what you're trying to verify, rather than using setting 305 from solution 99 to test priorities or whatever non-conventions we have used to date.   If it's not easy to do this, maybe leave a TODO describing what would need to be done.

Reply:
Hey @the-t-in-rtf  -- thanks for the comments!

Your use of scare quotes for '... "real" solutions ...' made me smile.  The solutions used here are only half real.  On one hand, the solutions registry is a mock, containing names like "Dummy solution registry entry".  The _real_ [solutions registries](https://github.com/GPII/universal/tree/master/testData/solutions) deployed in production are not involved.  On the other hand, the names of the solutions are real in the sense that there are such solutions in the real registries, e.g., `org.nvda-project`.

As for how difficult it is to switch over to completely mocked solutions, I don't think it's that difficult.  What's required for the browser channel tests is a coordinated set of:
- gpiiKey and associated prefsSafe (see comment below),
- the prefsSafe has preferences for one or more solutions in the SR,
- the device reporter declares that the solution is available,
- and the solution is referenced in the prefsSafe using "http://registry.gpii.net/applications/name.of.solution.here"

The gpiiKey/prefsSafes are generated from the test preferences documents in `%gpii-universal/tests/data/preferences/`, and the tests of the preferences payloads fetched via the browser channel is based on their contents.

What might be tricky is that these test definitions are shared with the integration tests, and that scenario is more real than these relatively simpler unit tests.  "One won't know until one tries".

Reply:
See my other comment, I think entangling the two is a double-edged sword.  The integration tests might not show us issues with the "real" data and give us the false impression that we've fixed something that we haven't.  Things that need real data need to use the SR directly.  Things that don't should be converted to use cleanly expressed synthetic SR entries.